Framework/API/src/InfoComponentVisitor.cpp

```diff
@@ -36,11 +58,14 @@ void InfoComponentVisitor::registerComponentAssembly(
     child->registerContents(*this);
   }
   const size_t detectorStop = m_assemblySortedDetectorIndices.size();
-
   m_ranges.emplace_back(std::make_pair(detectorStart, detectorStop));
 
   // For any non-detector we extend the m_componetIds from the back
   m_componentIds.emplace_back(assembly.getComponentID());
+
+  m_positions->emplace_back(Kernel::toVector3d(assembly.getPos()));
+  m_rotations->emplace_back(Kernel::toQuaterniond(assembly.getRotation()));
+  clearPositionAndRotationsParameters(m_pmap, assembly);
```

Comment:
I'm not 100% sure what your concern is here. But the parameter map will be cleared depth first. So any parent information will always still be in the parameter map at the point that child (derived) information is fetched. 

Probably having a test for this is still a good idea.